File: TransCoder_tokenized_test_set_functions/python/EXPONENTIAL_SQUARING_FAST_MODULO_MULTIPLICATION_1.py
def exponentiation ( bas, exp ) :
    t = 1;
    
    while ( exp > 0 ) :
        if ( exp % 2 != 0 ) :
            t = ( t * bas ) % N;
            
        bas = ( bas * bas ) % N;
        
        exp = int ( exp / 2 );
        
    return t % N;

File: results/java_codex/EXPONENTIAL_SQUARING_FAST_MODULO_MULTIPLICATION_1.java
long exponentiation ( long base, long exp ) {
  long t = 1L;
  while ( exp > 0 ) {
    if ( exp % 2 != 0 ) t = ( t * base ) % N;
    base = ( base * base ) % N;
    exp /= 2;
  }
  return t % N;
}